lib/puppet/provider/user/windows_adsi.rb

```diff
@@ -125,14 +125,13 @@ def password
 
   def password=(value)
     if user.disabled?
-      warning _("The user account '%s' is disabled; puppet will not reset the password" % @resource[:name])
+      warning _("The user account '%s' is disabled; The password will still be changed" % @resource[:name])
     elsif user.locked_out?
-      warning _("The user account '%s' is locked out; puppet will not reset the password" % @resource[:name])
+      warning _("The user account '%s' is locked out; The password will still be changed" % @resource[:name])
     elsif user.expired?
-      warning _("The user account '%s' is expired; puppet will not reset the password" % @resource[:name])
-    else
-      user.password = value
+      warning _("The user account '%s' is expired; The password will still be changed" % @resource[:name])
     end
+      user.password = value
```

Comment:
This doesn't seem right to me. Puppet can't manage the password if the account is disabled, since it would need to call `LogonUser`. I left more comments in the ticket.

Reply:
I talked with @mihaibuzgau about this. It's confusing due to the various changes over the years. I didn't get a chance to look at it today, but will tomorrow.

Reply:
visual: too much indentation here
```suggestion
    user.password = value
```